dags/common/enricher.py

```diff
@@ -0,0 +1,111 @@
+import xml.etree.ElementTree as ET
+from typing import Dict
+
+import backoff
+import requests
+from structlog import get_logger
+
+
+class Enricher(object):
+    def __init__(self) -> None:
+        self.logger = get_logger().bind(class_name=type(self).__name__)
+
+    def _get_schema(self):
+        return "http://repo.scoap3.org/schemas/hep.json"
+
+    @backoff.on_exception(
+        backoff.expo, requests.exceptions.RequestException, max_time=60, max_tries=3
+    )
+    def _get_arxiv_categories(self, arxiv_id=None, title=None, doi=None):
+        if arxiv_id is None and title is None and doi is None:
+            raise ValueError(
+                "One of the arxiv_id, title and doi parameters has to be different then None."
+            )
+
+        def clean_arxiv(arxiv):
+            if arxiv is None:
+                return None
+
+            return arxiv.split(":")[-1].split("v")[0].split(" ")[0].strip("\"'")
+
+        def get_arxiv_categories_from_response_xml(xml: ET.Element):
+            xml_namespaces = {
+                "arxiv": "http://arxiv.org/schemas/atom",
+                "w3": "http://www.w3.org/2005/Atom",
+            }
+
+            entries = xml.findall("./w3:entry", namespaces=xml_namespaces)
+            if len(entries) != 1:
+                return []
+            entry = entries[0]
+
+            primary_categories = [
+                node.attrib["term"]
+                for node in entry.findall(
+                    "./arxiv:primary_category", namespaces=xml_namespaces
+                )
+            ]
+            if not primary_categories:
+                return []
+            if len(primary_categories) > 1:
+                self.logger.error(
+                    f"Arxiv returned {len(primary_categories)} primary categories."
+                )
+                return []
+            primary_category = primary_categories[0]
+
+            secondary_categories = [
+                node.attrib["term"]
+                for node in entry.findall("./w3:category", namespaces=xml_namespaces)
+            ]
+
+            return list(set([primary_category] + secondary_categories))
+
+        arxiv_id = clean_arxiv(arxiv_id)
+
+        search_query = []
+
+        if title:
+            title = (
+                title.replace("-", "?")
+                .encode("ascii", "replace")
+                .decode()
+                .replace("%22", "+")
```

Comment:
why here need to replace %22 (which is ") to +?

Reply:
Yes good point, was a leftover from tests